server/handler.go

```diff
@@ -0,0 +1,55 @@
+package server
+
+import "encoding/base64"
+
+func (server *ConiksServer) handleRegistrationMessage(reg *RegistrationRequest) (Response, error) {
+	if len(reg.Username) == 0 ||
+		len(reg.KeyType) == 0 ||
+		len(reg.Key) == 0 {
+		return NewErrorResponse(ErrorMalformedClientMessage),
+			ErrorMalformedClientMessage
+	}
+
+	// decode key string
+	key, err := base64.StdEncoding.DecodeString(reg.Key)
+	if err != nil {
+		return NewErrorResponse(ErrorMalformedClientMessage),
+			ErrorMalformedClientMessage
+	}
+
+	server.state.RLock()
+	// check whether the name already exists
```

Comment:
I think L21-45 could be moved to a ``Register(name, key, db)`` function in the ``protocols`` package.